pkg/engine/events.go

```diff
@@ -428,6 +442,47 @@ func (e *eventEmitter) updateSummaryEvent(maybeCorrupt bool,
 	}
 }
 
+func (e *eventEmitter) policyViolationEvent(urn resource.URN, d plugin.AnalyzeDiagnostic) {
+
+	contract.Requiref(e != nil, "e", "!= nil")
+
+	// Write prefix.
```

Comment:
I don't love this code, which is spiritually a lot like `diag.Sink`. I don't think it's right to put it on the `AnalyzeDiagnostic` though.